contrib/xor-blocks/src/main.rs

```diff
@@ -0,0 +1,130 @@
+use std::{
+    env, fs,
+    io::{self, Read},
+    path::Path,
+    time::{Duration, Instant},
+};
+
+const MAGIC: [u8; 4] = [0xf9, 0xbe, 0xb4, 0xd9];
+
+fn main() -> Result<(), io::Error> {
+    let start = Instant::now();
+
+    let blocks_path = if env::args().len() > 1 {
+        env::args().nth(1).expect("the arg to exist").into()
```

Comment:
Consider accepting this argument in the same way it's specified with `bitcoind` and `bitcoin-cli`, that is, `-datadir=<dir>`. Besides being consistent with these existing executables, it would also be cleaner if more arguments were added in the future (because the data directory wouldn't be specified in a different way from other arguments).

Reply:
Sadly, that's not really that easy because of damn Windows non-compliant UTF-16. I wrote a crate for it called `parse_arg` but it's soft-deprecated there. Feel free to copy-paste or just depdend on it. Maybe I'll end up putting it into another dedicated crate.

But what I'd like to see here is having proper usage page if no arguments are provided. `clap` or `configure_me` can do that but for a simple tool a bit of manual code is likely fine.

Reply:
I added checking for `-datadir=` and `--datadir=`, but not `datadir=` or `---datadir=`. Just like `bitcoind` :).

> I'd like to see here is having proper usage page if no arguments are provided

If no args are provided, it just picks up the default datadir. IMO that's the best behavior. If you give more than 1 arg it prints out a usage instruction as an error.
Is that what you had in mind?